dso_pkg/dso/policy_optimizer/policy_optimizer.py

```
@@ -189,10 +189,15 @@ def _apply_gradients(self, gradients):
             zip(gradients, self.policy.controller.trainable_variables)
         )
 
-    @abstractmethod
     def _compute_loss(self, obs, actions, rewards):
-        """Compute the loss function. Must be implemented by subclasses."""
-        pass
+        """Compute the loss function. Default implementation."""
+        # Default implementation using policy gradient approach
+        with tf.GradientTape() as tape:
```

Ruff (0.12.2)

Local variable `tape` is assigned to but never used

Remove assignment to unused variable `tape`

(F841)